infoschema/slow_log.go

```diff
@@ -64,12 +64,32 @@ var slowQueryCols = []columnInfo{
 }
 
 func dataForSlowLog(ctx sessionctx.Context) ([][]types.Datum, error) {
-	return parseSlowLogFile(ctx.GetSessionVars().Location(), ctx.GetSessionVars().SlowQueryFile)
+	return ReadSlowLogData(ctx.GetSessionVars().SlowQueryFile, ctx.GetSessionVars().Location())
 }
 
-// parseSlowLogFile uses to parse slow log file.
+// ReadSlowLogData uses to read slow log data.
+// It is exporting for testing.
 // TODO: Support parse multiple log-files.
-func parseSlowLogFile(tz *time.Location, filePath string) ([][]types.Datum, error) {
+func ReadSlowLogData(filePath string, tz *time.Location) ([][]types.Datum, error) {
+	return globalSlowQueryReader.getSlowLogData(filePath, tz)
+}
+
+// ParseSlowLogRows reads slow log data by parse slow log file.
+// It won't use the cache data.
+// It is exporting for testing.
+func ParseSlowLogRows(filePath string, tz *time.Location) ([][]types.Datum, error) {
```

Comment:
Since we have `ReadSlowLogData` now, why not remove this function?

Reply:
`ParseSlowLogRows` only use to test and `ParseSlowLogRows` won't update cache.
`ReadSlowLogData` will update cache, but some test only wanto test parse slow log.